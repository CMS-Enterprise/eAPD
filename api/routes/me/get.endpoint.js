const {
  getDB,
  getFullPath,
  login,
  request,
  unauthenticatedTest
} = require('../../utils.endpoint');

const url = getFullPath('/me');

describe('/me endpoint | GET', () => {
  const db = getDB();
  beforeAll(() => db.seed.run());
  afterAll(() => db.destroy());

  unauthenticatedTest('get', url);

  it('when authenticated', async () => {
    const cookies = await login();
    const { response, body } = await request.get(url, {
      jar: cookies,
      json: true
    });

    expect(response.statusCode).toEqual(200);
    expect(body).toMatchSnapshot();
  });
});
