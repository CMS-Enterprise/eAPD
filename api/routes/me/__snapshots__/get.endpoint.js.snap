// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/me endpoint | GET when authenticated 1`] = `
Object {
  "activities": Array [
    "view-users",
    "add-users",
    "view-roles",
    "create-roles",
    "edit-roles",
    "submit-federal-response",
    "submit-clearance",
    "edit-comments",
    "submit-document",
    "submit-state-response",
    "create-draft",
    "edit-document",
    "edit-response",
    "delete-users",
    "view-document",
  ],
  "id": 2000,
  "role": "admin",
  "state": Object {
    "id": "mn",
    "name": "Minnesota",
  },
  "username": "all-permissions-and-state",
}
`;
